draft-ietf-sidrops-publication-server-bcp-01.md

```diff
@@ -139,6 +139,70 @@ of the root cause for disruption in the Publication Server that effectively is
+publication protocol. If this protocol is not available then this will prevent
+publishers from updating their ROAs and renewing their manifests and CRLs.
+
+Publication Servers MUST aim to minimise data loss in case of severe server
+outages. If a server restore is needed and a content regression has occured due
+to data loss then the server MUST perform an RRDP session reset.
+
+Publishing CAs typically only check in with their Publication Server when they
+have changes that need to be published. As a result they may not be aware if the
+server performed a restore and their content regressed to an earlier state. This
+could result in two problems:
+
+ - The ROAs may not reflect what the publisher intended
+ - The publisher may not renew their manifest or CRL in time, because they
+   assume that their current manifest and CRL have not yet expired or become
+   stale
+
+Therefore, the Publication Server SHOULD notify publishing CAs about this issue
+if it occurs, so that a full manually triggered resynchronisation can then be
+initiated by CAs.
+
+## Publisher Repository Synchronisation
+
+### Full Synchronisation
+
+It is RECOMMENDED that publishing CAs always perform a list query as described
+in section 2.3 of [@!RFC8181] before sending all their changes using multiple
+PDUs as described in section 2.2 of [@!RFC8181]. This way any desynchronisation
+issue can be resolved at least as soon as the publisher is aware of updates that
+it needs to publish.
+
+In addition to this the publishing CA MAY perform regular planned
+synchronisation events where it issues an [@!RFC8181] list query even if it has
+no new content to publish. Because this interaction requires that the
+Publication Server signs an [@!RFC8181] list reply, this operation can be costly
+for Publication Servers that serve a large number of publishers. Therefore,
+publishers SHOULD NOT perform this resynchronisation more frequently than once
+every 10 minutes.
+
+### Monitoring Repository Content
+
+It should be noted that the [@!RFC8181] list reply returned by a Publication
+Server can differ from the actual set of objects that are published. This is
+because the Publication Server needs time between accepting updates from one
+of their publishers and updating the content of the public repositories, or it
+may be because the Publication Server chooses to limit the number of deltas that
+it publishes to a maximum of 1 delta per minute as described in section 5.4 of
+this document.
+
+In either case, the publishing CA can monitor the content of their Publication
+Server's RRDP repository to verify that the objects it expects are updated in
+a timely manner. Because this would rely on conditional retrieval of the RRDP
+Notification file the CA MAY perform this verification every minute.
+
+If the expected files are not found to be published within a reasonable time
+(let's say 5 minutes?), or if the CA recognises that there is a regression in
```

Comment:
I'm not sure about a suggestion like this.  There will typically be a few moving parts between the publication server and the RRDP service accessed by the client, and it may not be that often that repeating the updates will fix the problem.  Maybe instead something like "expect to see objects within five minutes, and if you don't, please contact the publication service operator"?

Reply:
As mentioned above for context there, my idea was that hammering the CDN every minute would be fine.

- wrt published within a reasonable time

But, you are right of course. There are moving parts in between and the delay can even be 10 minutes or longer dependent on setup. And, unfortunately (another thing for my publication++ wishlist) there is no indication to the client about how long is normal here (e.g. the RFC8181 success reply could have included a hint).

So, if we include a time here then we should probably be conservative and use something like 15 minutes? But note, that if the publisher does a full resync after 5 minutes, this is probably not an issue. It would just be another list request with a reply that tells the publisher that everything is there.

- wrt CA recognises... regression

I think there is merit in monitoring the repository for changes regardless of recent publication activity. It could help publishers discover that the repository has *regressed*, in which cast "contact publication server operator" could just be, issue a warning and do another full synchronisation (RFC8181 list and publish diff).


Does this make sense? I will think about updated text, but suggestions are welcome of course.

Reply:
> So, if we include a time here then we should probably be conservative and use something like 15 minutes?

I think 5 minutes is better, because in the normal course of things that should be the outer limit of any delay.  If it's regularly taking longer than that, then the setup on the server side needs to be improved.

> But note, that if the publisher does a full resync after 5 minutes, this is probably not an issue. It would just be another list request with a reply that tells the publisher that everything is there.

Yep, that's a fair point.  (I'd assumed that 'resynchronisation' meant 'delete existing objects and re-upload', which I think can happen inside a single request, so it may be worth documenting the resynchronisation process.)

Reply:
Can you please check if this is improved by [this](https://github.com/timbru/draft-sidrops-publication-server-bcp/pull/30/commits/32ce651bd9bd746442256eaa72f78d7bd1370201).

Reply:
Sorry, after looking at this again I realised that you had documented
resynchronisation separately in the original text.  I would clarify
that existing section (second paragraph of 4.4.1) like so:

```
In addition to this, the publishing CA MAY verify periodically that it
is synchronised with its publication server.  The CA does this by
issuing an [@!RFC8181] list query, and then issuing a query containing
the publish/withdraw PDUs that will cause the publication server to
have the correct repository state for the CA.  If the list query
indicates that the publication server has the correct state, then no
further action is required.  This process is termed "full
synchronisation".
```

with the later part becoming:

```
If the expected files are not found to be published within 5
minutes, or a regression is observed, then the CA SHOULD perform full
resynchronisation per section 4.4.1.  If the list reply received
during full resynchronisation indicates that the publication server
has the intended repository state for the CA, then this points at an
operational issue at the publication server.
```